include/bio/map_io/reader_options.hpp

```diff
@@ -0,0 +1,119 @@
+// -----------------------------------------------------------------------------------------------------
+// Copyright (c) 2006-2021, Knut Reinert & Freie Universität Berlin
+// Copyright (c) 2016-2021, Knut Reinert & MPI für molekulare Genetik
+// Copyright (c) 2020-2021, deCODE Genetics
+// This file may be used, modified and/or redistributed under the terms of the 3-clause BSD-License
+// shipped with this file and also available at: https://github.com/seqan/seqan3/blob/master/LICENSE.md
+// -----------------------------------------------------------------------------------------------------
+
+/*!\file
+ * \brief Provides bio::map_io::reader_options and various pre-defined field_types.
+ * \author Hannes Hauswedell <hannes.hauswedell AT decode.is>
+ */
+
+#pragma once
+
+#include <string>
+#include <string_view>
+#include <vector>
+
+#include <seqan3/alphabet/nucleotide/dna5.hpp>
+#include <seqan3/alphabet/quality/phred42.hpp>
+#include <seqan3/alphabet/views/char_to.hpp>
+#include <seqan3/utility/type_list/traits.hpp>
+
+#include <bio/format/sam.hpp>
+#include <bio/map_io/header.hpp>
+#include <bio/map_io/misc.hpp>
+#include <bio/map_io/sam_flag.hpp>
+#include <bio/map_io/sam_tag_dictionary.hpp>
+#include <bio/stream/transparent_istream.hpp>
+#include <bio/stream/transparent_ostream.hpp>
+
+namespace bio::map_io
+{
+
+/*!\name Pre-defined field types
+ * \brief These can be used to configure the behaviour of the bio::map_io::reader va bio::map_io::reader_options.
+ * \{
+ */
+
+/*!\brief The default field types for variant io.
+ *!\ingroup map_io
+ *
+ * \details
+ *
+ * These traits define a record type with minimal memory allocations for all input formats.
+ * It is the recommended record type when iterating ("streaming") over files that ca be any variant IO format.
+ *
+ * The "style" of the record resembles the BCF specification, i.e. contigs, FILTERs and INFO identifiers are
+ * represented as numbers (not strings); and the genotypes are encoded by-genotype (not by-sample).
+ * See bio::map_io::genotypes_bcf_style for more information on the latter.
+ *
+ * \warning Shallow types
+ *
+ * These records are not self-contained, i.e. they depend on caches and will become invalid when the reader moves to
+ * the next record.
+ * Since some elements in the record are views, it may not be possible and/or safe to change all values.
+ */
+template <ownership own = ownership::shallow>
+inline constexpr auto field_types_sam =
```

Comment:
```suggestion
inline constexpr auto field_types =
```
I decided to simplify this strongly in VCF/BCF I/O. I recommend against using "format"-optimised field_types and just call these the default.